File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        unordered_map<int, int>::iterator m = mp.end();
        int t2, i;
        for (i = 0; i < nums.size(); ++i) {
            t2 = target - nums[i];
            if (mp.find(t2) != m) {
                break;
            } else mp[nums[i]] = i;
        }
        return { mp[t2], i };
    }
};

File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val mp : HashMap<Int, Int> = HashMap<Int, Int>();
        var k : Int = 0;
        var i : Int = 0;
        for ((idx, n) in nums.withIndex()) {
            k = target - n;
            i = idx;
            if (mp[k] != null) 
                break;
            mp.put(n, idx);
        }
        return intArrayOf(mp[k] ?: 0, i);
    }
}